book/views.py

```diff
@@ -1,3 +1,25 @@
-from django.shortcuts import render
+import json
+from datetime         import timedelta, date
 
-# Create your views here.
+from django.views     import View
+from django.http      import JsonResponse
+
+from .models          import Book
+
+
+class RecentlyBookView(View):
+    def get (self, request):
+
+        today          = date.today()
+        previous_days  = today - timedelta(days=30)
+
+        books = Book.objects.filter(publication_date__range=
+                                    [previous_days,today]).order_by('-publication_date')[:20]
```

Comment:
```suggestion
                                    [previous_days,today]).order_by('-publication_date')[:limit]
```
한 번에 모든 데이터를 불러 오지 않고 범위를 지정하는 것은 굉장히 좋은 습관입니다 👍 
보통 이를 limit이라 지칭하는데요, 프론트엔드로 부터 offset과 limit을 쿼리스트링으로 받아오면 좀 더 유동적인 api가 될 수 있습니다. 쿼리스트링이 없을 때는 default를 20으로 지정해두시면 원래 의도하신대로 적용이 가능합니다 :)

Reply:
쿼리스트링 처리로 수정했습니다. 감사합니다.